File: php/features.php
<?php
declare(strict_types=1);

// ArulsIp SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class ArulsIpFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new ArulsIpBaseFeature();
            case "test":
                return new ArulsIpTestFeature();
            default:
                return new ArulsIpBaseFeature();
        }
    }
}

File: py/features.py
# ArulsIp SDK feature factory

from feature.base_feature import ArulsIpBaseFeature
from feature.test_feature import ArulsIpTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ArulsIpBaseFeature(),
        "test": lambda: ArulsIpTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
